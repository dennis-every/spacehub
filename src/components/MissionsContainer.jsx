import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissions } from '../redux/missions/missionsSlice';
import Mission from './Mission';

const MissionsContainer = () => {
  const dispatch = useDispatch();
  const ifSucceed = useSelector((store) => store.missions.ifSucceed);
  const isLoading = useSelector((store) => store.missions.isLoading);
  const missions = useSelector((store) => store.missions.missions);

  useEffect(() => {
    dispatch(fetchMissions());
  }, [ifSucceed, dispatch]);

  let content;
  if (isLoading) {
    content = (
      <tr>
        <td>Loading...</td>
      </tr>
    );
  } else if (ifSucceed) {
    content = missions.map((mission) => (
      <Mission
        key={mission.mission_id}
        id={mission.mission_id}
        name={mission.mission_name}
        description={mission.description}
        status={mission.status}
      />
    ));
  } else {
    content = (
      <tr>
        <td>Something went wrong...</td>
      </tr>
    );
  }

  return (
    <table className="table table-striped table-bordered">
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th> </th>
        </tr>
      </thead>
      <tbody>{content}</tbody>
    </table>
  );
};

export default MissionsContainer;
