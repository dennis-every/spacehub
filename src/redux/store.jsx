import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketsReducer,
  },
});

export default store;
