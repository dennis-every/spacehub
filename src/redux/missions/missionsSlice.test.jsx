import { render, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import MissionsContainer from '../../components/MissionsContainer';
import store from '../store';

const mockStore = configureStore([]);

describe('Missions redux state tests', () => {
  it('Expect to initially set missions to an empty object', () => {
    const state = store.getState().missions;
    const initialState = {
      missionsArray: [], ifSucceed: false, isLoading: false, errors: null,
    };
    expect(state).toEqual(initialState);
  });
  it('Expect to display one mission from mockStore', () => {
    const missionsArray = [
      {
        mission_name: 'Thaicom',
        mission_id: '9D1B7E0',
        description:
          'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
        reserved: true,
      },
    ];
    const state = {
      missions: {
        missionsArray,
        ifSucceed: true,
        isLoading: false,
        errors: null,
      },
    };
    const store = mockStore(state);
    render(
      <Provider store={store}>
        <MissionsContainer />
      </Provider>,
    );
    const missionName = screen.getByText('Thaicom');
    expect(missionName).toBeInTheDocument();
    expect(store.getState).toMatchSnapshot();
  });
});
