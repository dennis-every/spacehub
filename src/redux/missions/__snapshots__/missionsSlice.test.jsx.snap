// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions redux state tests Expect to display one mission from mockStore 1`] = `[Function]`;
